File: FidelityHub.Web/ClientApp/src/app/shared/models/promotion.ts
import { VendorUnit } from "../../vendor/models/vendor";

export class Promotion {
  id: number;
  startDate: Date;
  endDate: Date;
  promotionHourStart: Date;
  promotionHourEnd: number;
  vendorUnit: VendorUnit;
  promotionType: PromotionType;
  description: string;
}

export class PromotionType {
  id: number;
  description: string;
  threshold: number;
}

export class PromotionSale {
  id: number;
  promotion: Promotion;
  userId: string;
  timestamp: Date;
  amount: number;
  approved: boolean;
  removed: boolean;
}

File: FidelityHub.Database/Entities/AppSchema/Promotion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace FidelityHub.Database.Entities.AppSchema
{
    public class Promotion
    {
        [DatabaseGeneratedAttribute(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int PromotionHourStart { get; set; }
        public int PromotionHourEnd { get; set; }
        public string Description { get; set; }
        public bool Active { get; set; }
        //FKs
        public int VendorUnitId { get; set; }
        public VendorUnit VendorUnit { get; set; }
        public int PromotionTypeId { get; set; }
        public PromotionType PromotionType { get; set; }
    }
}
